webapp/src/app.tsx

```diff
@@ -1332,7 +1332,8 @@ export class ProjectView
                                 // We are too late; the editor has already been loaded.
                                 // Call the onChanges handler to update the editor.
                                 pxt.tickEvent(`identity.syncOnProjectOpen.timedout`, { 'elapsedSec': elapsed})
-                                cloud.onChangesSynced(changes)
+                                if (changes.length)
+                                    cloud.forceReloadForCloudSync()
```

Comment:
Can this call trigger a reload while in the editor? I've seen the toast "Cloud synchronization complete. Reloading" happen in the editor, which was jarring. Logged it here: https://github.com/microsoft/pxt-arcade/issues/3005

Reply:
Yes, it can. Unfortunately I don't know a good way to avoid this.

The problem is that if the user opens up a project at version A but after that we sync with the cloud and determine they should be fast-forwarded to version B, then we need to reload the editor from underneath them. Currently, there isn't a way to reload the editor's content without doing a page reload. Likely we could fix this using virtual APIs but it's another chunk of work..